### PR TITLE
Fix the headless simulator

### PR DESCRIPTION
**Pull request by trotod**
"Shims" `lscache`, support io.js < v3.0.0. Make simulating `one-game` possible.

Tested on Linux with Node.js v0.10.40 and v0.12.7; as well as io.js v2.5.0.

It still won't work with io.js v3.0.0 because `webworker-threads` doesn't compile properly, so that's another problem.

Hopefully nothing else changed (as in the actual simulation results).

### CODE REVIEW

File: headless_client/worker_world.coffee [hunk 1]
```diff
@@ -191,11 +189,11 @@ for codeFile in [
     'app/vendor/aether-lua.js'
     'app/vendor/aether-python.js'
   ]
-  codeFileContents.push fs.readFileSync("./public/javascripts/#{codeFile}", 'utf8')
+  codeFileContents.push fs.readFileSync(__dirname + "/../public/javascripts/#{codeFile}", 'utf8')
 
 #window.BOX2D_ENABLED = true;
 
-newConsole = "newConsole = #{}JASON.stringify newConsole}()";
+newConsole = "newConsole = #{}(newConsole.toString())}()"
```
Comment (trotod) [inline headless_client/worker_world.coffee]: Also, is this line even necessary? It's not used anywhere, it doesn't look right, and `newConsole` was never defined earlier...

Comment (nwinter) [inline headless_client/worker_world.coffee]: ... nope!

**Comment by nwinter** [comment]
Thanks @trotod! As much as I love bug reports, I love pull requests that fix bugs I didn't even know about way more. :D

**Comment by trotod** [comment]
Oh wait...I forgot to update `request`. :disappointed:

Without the latest `request` module the simulator on Node v0.12 and io.js for some reason only is able to simulate one game. It still works fine on Node v0.10.

It's being used in another files, so I'm not sure if it's safe to update it; though it's only a minor version upgrade.

**Comment by nwinter** [comment]
Looks like there have been a ton of changes in between request 2.12 and 2.60, and it doesn't follow semver or list breaking changes separately, so we'd want to run the tests and also thoroughly prod the site before updating that.

**Pull request merged**